test-engine-core-modules/src/rmodel/rmodel.py

```diff
@@ -0,0 +1,199 @@
+    _metadata: PluginMetadata = PluginMetadata(_name, _description, _version)
+    _plugin_type: PluginType = PluginType.MODEL
+    _model_plugin_type: ModelPluginType = ModelPluginType.R
+
+    @staticmethod
+    def get_metadata() -> PluginMetadata:
+        """
+        A method to return the metadata for this plugin
+
+        Returns:
+            PluginMetadata: Metadata of this plugin
+        """
+        return Plugin._metadata
+
+    @staticmethod
+    def get_plugin_type() -> PluginType:
+        """
+        A method to return the type for this plugin
+
+        Returns:
+             PluginType: Type of this plugin
+        """
+        return Plugin._plugin_type
+
+    @staticmethod
+    def get_model_plugin_type() -> ModelPluginType:
+        """
+        A method to return ModelPluginType
+
+        Returns:
+            ModelPluginType: Model Plugin Type
+        """
+        return Plugin._model_plugin_type
+
+    def __init__(self, model: Any) -> None:
+        if model:
+            self._model = model
+
+    def cleanup(self) -> None:
+        """
+        A method to clean-up objects
+        """
+        pass  # pragma: no cover
+
+    def setup(self) -> Tuple[bool, str]:
+        """
+        A method to perform setup
+
+        Returns:
+            Tuple[bool, str]: Returns bool to indicate success, str will indicate the
+            error message if failed.
+        """
+        is_success = True
+        error_messages = ""
+        return is_success, error_messages
+
+    def get_model(self) -> Any:
+        """
+        A method to return the model
+
+        Returns:
+            Any: Model
+        """
+        if self._model:
+            return self._model
+        else:
+            return None
+
+    def get_model_algorithm(self) -> str:
+        """
+        A method to return the model algorithm.
+        Either one of the supported algorithms or ""
+
+        Returns:
+            str: model algorithm name if supported or "" if not supported
+        """
+        return self._model_algorithm
+
+    def is_supported(self) -> bool:
+        """
+        A method to check whether the model is being identified correctly
+        and is supported
+
+        Returns:
+            bool: True if is an instance of model and is supported
+        """
+        is_algorithm_supported, model_algorithm = self._identify_model_algorithm(
+            self._model
+        )
+
+        if is_algorithm_supported:
+            self._model_algorithm = model_algorithm
+        else:
+            # Not supported
+            pass
+
+        return is_algorithm_supported
+
+    def predict(self, data: Any, *args) -> Any:
+        """
+        A method to perform prediction using the model
+
+        Args:
+            data (Any): data to be predicted by the model
+
+        Returns:
+            Any: predicted result
+        """
+
+        try:
+            if isinstance(data, list):
+                for item in data:
```

Comment:
Hey there. Just wondering in a scenario where the data is an empty list, would you like to return an empty np.array?

Reply:
Hmm no I would throw an error that it is not a model to be honest, I looked a bit at the sklearnmodel.py and there it is implemented like this, so I opted for the being consistent instead of correct

Reply:
ok this is fine. thanks!